File: Leetcode/Python/_452.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        if not points:
            return 0

        points.sort(key=lambda x: x[1])

        arrows = 1
        first_end = points[0][1]
        for start, end in points:
            if first_end < start:
                arrows += 1
                first_end = end
                
        return arrows

File: Leetcode/cpp/452.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        std::sort(
            points.begin(), points.end(),
            [](vector<int> a, vector<int> b) {
                if (a[1] < b[1]) {
                    return true;
                }
                return false;
            }
        );
        int counter = 1;
        int position = points[0].back();
        for (int i = 1; i < points.size(); ++i) {
            if (position < points[i][0]) {
                position = points[i][1];
                ++counter;
            }
        }

        return counter;
    }
};
